OPHD/UI/Core/ToolTip.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "Control.h"
+
+#include <NAS2D/Renderer/Point.h>
+#include <NAS2D/Resources/Font.h>
+#include <NAS2D/Timer.h>
+
+#include <utility>
+#include <vector>
+
+
+/**
+ * \class Label
+ * \brief A control that contains readonly text.
+ *
+ */
+class ToolTip : public Control
+{
+public:
+	ToolTip();
+	virtual ~ToolTip();
+
+	void add(Control&, const std::string&);
+	void update() override;
+
+private:
+	void onMouseMove(int, int, int, int);
+
+private:
+	const NAS2D::Font& mFont;
+	NAS2D::Timer mTimer;
+
+	NAS2D::Point<int> mMouseCoords;
+
+	std::pair<Control*, std::string>* mFocusedControl;
+
+	std::vector<std::pair<Control*, std::string>> mControls;

```

Comment:
The resulting code would probably be a little more readable if you defined a `struct` to hold `Control` and `std::string` values as named fields, rather than as `std::pair` with values `first` and `second`.

Reply:
Why build anew when I can instead use what's available in the standard library?

I was using `typedef`'s way back when and eventually `using` directives but those mostly got stripped out because of this kind of reasoning. I dunno, I find that defining a `struct` or `class` just for this sort of case when I could use a `std::pair` or `std::tuple` instead to be overkill.

Reply:
See the second answer, which has the most up votes:
https://stackoverflow.com/questions/3607352/struct-with-2-cells-vs-stdpair

This was also address in The Clean Code Collection, which also suggested using a custom `struct` to improve readability:
https://www.amazon.com/Robert-Martin-Clean-Code-Collection-ebook/dp/B00666M59G

Reply:
Honestly that's a pretty good answer and a strong argument against std::pair/std::tuple. Could be something to look at going forward. Plus, to be fair, if needs change for elements later on it's a trivial matter to add an additional member type.